jerry-core/parser/js/js-parser-statm.c

```diff
@@ -1656,6 +1656,21 @@ parser_parse_statements (parser_context_t *context_p) /**< context */
     JERRY_ASSERT (context_p->stack_depth == context_p->context_stack_depth);
 #endif /* !JERRY_NDEBUG */
 
+    /* Add breakpoints, before every single line. We have two type of breakpoints
+    *  one of them the active (enabled) others the inactive (disabled). In basic
+    *  state the breakpoints is desabled. When the additional breakpoint is active,
+    *  the engine stop running on the breakpoint. The other option is when the
+    *  breakpoint is inactive and there is nothing to do when the engine step
+    *  in the breakpoint.
+    */
+#ifdef JERRY_DEBUGGER
+    if (JERRY_CONTEXT (jerry_init_flags) & JERRY_INIT_DEBUGGER)
+    {
+      parser_emit_cbc (context_p, CBC_BREAKPOINT_DISABLED);
+      parser_flush_cbc(context_p);
```

Comment:
It seems you add a breakpoint before each statement. You should only do it before each line according to the comment above.